pkg/csi/service/common/commonco/k8sorchestrator/k8sorchestrator.go

```diff
@@ -1502,6 +1504,54 @@ func (c *K8sOrchestrator) AnnotateVolumeSnapshot(ctx context.Context, volumeSnap
 	return c.updateVolumeSnapshotAnnotations(ctx, volumeSnapshotName, volumeSnapshotNamespace, annotations)
 }
 
+// IsVolumeSnapshotAnnotationPresent checks if the volumesnapshot CR has the given annotation
+// set or not.
+func (c *K8sOrchestrator) IsVolumeSnapshotAnnotationPresent(ctx context.Context, volumeSnapshotName string,
+	volumeSnapshotNamespace string, annotationKey string) (bool, error) {
+	log := logger.GetLogger(ctx)
+	retryCount := 0
+	isAnnotationFound := false
+	// TODO: interval and limit can be made configurable
+	interval := time.Second
+	limit := 2 * time.Minute
+	// Attempt to get VolumeSnapshot instance every second for 2minutes.
+	getVSErr := wait.PollImmediate(interval, limit, func() (bool, error) {
+		retryCount++
+		// Retrieve the volumesnapshot CR and verify that it exists
+		volumeSnapshot, err := c.snapshotterClient.SnapshotV1().VolumeSnapshots(volumeSnapshotNamespace).
+			Get(ctx, volumeSnapshotName, metav1.GetOptions{})
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				log.Errorf("attempt: %d, the volumesnapshot %s/%s is not found",
```

Comment:
we should return true right ?

Reply:
PollImmediate tries a condition func until it returns true, returns an error, or the timeout is reached.

We need to retry in case NotFound error occurred, hence need to return false, so that it retries.